release-content/0.14/migration-guides/11426_on_enter_startup_states.md

```diff
@@ -0,0 +1,81 @@
+In Bevy 0.13, the [`OnEnter`] schedules for states initialized via [`app.init_state`] would run after any systems in the `Startup` schedules.
+This is because [`apply_state_transitions`] was only run during the [`StateTransition`] schedule.
+
+This was a subtle bug: it was possible for the game to be in a particular state without having first *entered* it.
+Now, [`OnEnter`] state transition logic is handled immediately.
+See [bevy#13968](https://github.com/bevyengine/bevy/issues/13968) for more context on this decision.
+
+To migrate, choose one of the following options:
+
+1. Moving your startup systems to a state, as a variant of the state you're waiting for (e.g. `AppState::Setup`), and then transition out of it once the setup is complete.
+2. Moving your startup systems to a state, and making the other state a [sub state](https://github.com/bevyengine/bevy/blob/v0.14.0-rc.4/examples/state/sub_states.rs) that depends on the startup state's completion (e.g. `SetupState::SetupComplete`).
+
+Bevy 0.13:
```

Comment:
I'd prefer this and the rest of the code blocks to be merged into the following format, to be consistent with the rest of the guides:

```rust
// 0.13
[derive(States, Default)]
enum AppState {
    #[default]
    InMenu,
    InGame,
}

...

// 0.14 (Solution 1)
#[derive(States, Default)]
enum AppState {
    #[default]
    InMenu,
    InGame,
}

...

// 0.14 (Solution 2)
#[derive(States, Default)]
enum AppState {
    #[default]
    Setup
    InMenu,
    InGame,
}

...
```

(Ignore the ellipsis)